lib/trento/domain/host/host.ex

```diff
@@ -267,6 +270,27 @@ defmodule Trento.Domain.Host do
     }
   end
 
+  def execute(
+        %Host{host_id: nil},
+        %SelectHostChecks{}
+      ) do
+    {:error, :host_not_registered}
```

Comment:
I added this clause to keep the domain consistent. 
However, this will be likely deleted with deregistration https://github.com/trento-project/web/blob/deregistration/lib/trento/domain/host/host.ex#L127

Reply:
Indeed, it will cause conflicts